ProcessLib/GroundwaterFlowProcess.h

```diff
@@ -313,52 +312,51 @@ class GroundwaterFlowProcess : public Process
     {
         DBUG("Postprocessing GroundwaterFlowProcess.");
 
-#ifdef USE_PETSC // Note: this is only a test
-
-        std::vector<PetscScalar>  u(_x->size());
-        _x->getGlobalVector(&u[0]);
-
-        int rank;
-        MPI_Comm_rank(PETSC_COMM_WORLD, &rank);
-        if(rank == 0)
-        {
-            std::string output_file_name = file_name;
-            boost::erase_last(output_file_name, ".vtu");
-            output_file_name.append(".dat");
-            std::ofstream os(output_file_name);
-            os << "SCALARS HEAD double 1"<<endl;
-            os << "LOOKUP_TABLE default"<< endl;
-            for (std::size_t i = 0; i < u.size(); ++i)
-                 os << u[i] << "\n";
-            os.close();
-        }
-#else
-        std::string const property_name = "Result";
+        std::string const property_name = "Head [m]";
 
         // Get or create a property vector for results.
-        boost::optional<MeshLib::PropertyVector<double>&> result;
+        boost::optional<MeshLib::PropertyVector<ScalarType>&> result;
         if (_mesh.getProperties().hasPropertyVector(property_name))
         {
             result = _mesh.getProperties().template
-                getPropertyVector<double>(property_name);
+                getPropertyVector<ScalarType>(property_name);
         }
         else
         {
             result = _mesh.getProperties().template
-                createNewPropertyVector<double>(property_name,
+                createNewPropertyVector<ScalarType>(property_name,
                     MeshLib::MeshItemType::Node);
             result->resize(_x->size());
         }
         assert(result && result->size() == _x->size());
 
-        // Copy result
-        for (std::size_t i = 0; i < _x->size(); ++i)
-            (*result)[i] = (*_x)[i];
+        // If in the case of DDC under parallel computation
+        if ( (_x->getRangeEnd() - _x->getRangeEnd()) != _x->size())
+        {
+            std::vector<ScalarType>  u(_x->size());
```

Comment:
Yes. It is for every rank. For huge mesh, there will be another approach by dumping raw data of rank wise vector, and extract solution by a post-process as what I did in OGS5. Therefore, there will be two methods for parallel output  of solutions:
1. multiple vtu plus pvtu in this PR.
2. parallel raw local vector dumping into a single file (future work).